pkg/handler/enqueue_mapped.go

```diff
@@ -81,7 +82,12 @@ func (e *enqueueRequestsFromMapFunc[object, request]) Create(
 	q workqueue.TypedRateLimitingInterface[request],
 ) {
 	reqs := map[request]empty{}
-	e.mapAndEnqueue(ctx, q, evt.Object, reqs)
+	priorityQueue, isPriorityQueue := q.(priorityqueue.PriorityQueue[request])
```

Comment:
Here we only expose a constructor and not a type - Can we instead simply wrap the `enqueueRequestFromMapFunc` with `WithLowPriorityWhenUnchanged`?